restapi/src/main/java/io/stargate/web/resources/v2/schemas/UserDefinedTypesResource.java

```diff
@@ -0,0 +1,513 @@
+      })
+  public Response update(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam(HEADER_TOKEN_AUTHENTICATION)
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(value = "", required = true) @NotNull final UserDefinedTypeUpdate udtUpdate,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          AuthenticatedDB authenticatedDB =
+              db.getRestDataStoreForToken(token, getAllHeaders(request));
+
+          Keyspace keyspace = authenticatedDB.getDataStore().schema().keyspace(keyspaceName);
+          if (keyspace == null) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        "keyspace does not exists.", Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+
+          String typeName = udtUpdate.getName();
+          if (Strings.isNullOrEmpty(typeName)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        "Type name must be provided.", Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+
+          db.getAuthorizationService()
+              .authorizeSchemaWrite(
+                  authenticatedDB.getAuthenticationSubject(),
+                  keyspaceName,
+                  null,
+                  Scope.ALTER,
+                  SourceAPI.REST,
+                  ResourceKind.TYPE);
+
+          UserDefinedType udt =
+              ImmutableUserDefinedType.builder().keyspace(keyspaceName).name(typeName).build();
+
+          List<UserDefinedTypeField> fields = udtUpdate.getFieldDefinitions();
+          if (fields != null && !fields.isEmpty()) {
+            try {
+              List<Column> columns = getUdtColumns(keyspace, fields);
+              authenticatedDB
+                  .getDataStore()
+                  .queryBuilder()
+                  .alter()
+                  .type(keyspaceName, udt)
+                  .addColumn(columns)
+                  .build()
+                  .execute(ConsistencyLevel.LOCAL_QUORUM)
+                  .get();
+            } catch (IllegalArgumentException | InvalidRequestException ex) {
+              return Response.status(Response.Status.BAD_REQUEST)
+                  .entity(new Error(ex.getMessage(), Response.Status.BAD_REQUEST.getStatusCode()))
+                  .build();
+            }
+          } else if (udtUpdate.getRenameColumns() != null
+              && !udtUpdate.getRenameColumns().isEmpty()) {
```

Comment:
Should we flag an error if _both_ adds and renames are requested? Right now it looks like adds will be executed, but renames ignored. I guess it would also be fine to execute both adds first, then renames. WDYT?